### PR TITLE
As policy module

### PR DESCRIPTION
**Pull request by irina-pereiaslavskaia**

### CODE REVIEW

**Comment by irina-pereiaslavskaia** [comment]
recheck

**Comment by irina-pereiaslavskaia** [comment]
recheck

File: tests/sanity/ignore-2.9.txt [hunk 1]
```diff
@@ -2,37 +2,37 @@ plugins/modules/as_config.py validate-modules:missing-gplv3-license
 plugins/modules/as_config_info.py validate-modules:missing-gplv3-license
```
Comment (anton-sidelnikov) [inline tests/sanity/ignore-2.9.txt:2]: rollback this file

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: plugins/modules/as_policy.py [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+        scheduled_policy = self.params['scheduled_policy']
+        scaling_policy_action = self.params['scaling_policy_action']
+        cool_down_time = self.params['cool_down_time']
+        state = self.params['state']
+
+        attrs = {}
+
+        changed = False
+
+        if as_group:
```
Comment (anton-sidelnikov) [inline plugins/modules/as_policy.py:397]: too deep make it more plain

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: plugins/modules/as_policy.py [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+
+        launch_time = scheduled_policy['launch_time']
+        recurrence_type = scheduled_policy['recurrence_type']
+        recurrence_value = scheduled_policy['recurrence_value']
+        start_time = scheduled_policy['start_time']
+        end_time = scheduled_policy['end_time']
+
+        sc_policy = {}
+
+        if launch_time:
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply plugins/modules/as_policy.py:275]: Done

File: plugins/modules/as_policy.py [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+    alarm: "as-alarm-test"
+  register: result
+
+# Update AS Policy
+- opentelekomcloud.cloud.as_policy:
+    scaling_group: "as-group-test"
+    scaling_policy: "collection-test"
+    scaling_policy_type: "alarm"
+    alarm: "as-alarm-test"
+    state: "present"
```
Comment (lego963) [inline plugins/modules/as_policy.py:173]: is `state` required param in update?

**Review by lego963 — CHANGES_REQUESTED** [review]

File: plugins/modules/as_policy.py (hunk 4 above)
Comment (irina-pereiaslavskaia) [reply plugins/modules/as_policy.py]: Done

File: plugins/modules/as_policy.py (hunk 2 above)
Comment (irina-pereiaslavskaia) [reply plugins/modules/as_policy.py:540]: Done

**Review by anton-sidelnikov — APPROVED** [review]

**Review by lego963 — APPROVED** [review]

**Pull request merged**